src/config.rs

```diff
@@ -0,0 +1,327 @@
+    ///  - `id`=`<ID>` a config's ID
+    ///  - `label`=`key` or `label`=`"key=value"` of a config label
+    ///  - `name`=`<name>` a config's name
+    ///  - `names`=`<name>` a multiple config's name comma separated
+    #[serde(serialize_with = "crate::docker::serialize_as_json")]
+    pub filters: HashMap<T, Vec<T>>,
+}
+
+impl<T> From<ListConfigsOptions<T>> for crate::query_parameters::ListConfigsOptions
+where
+    T: Into<String> + Eq + Hash + serde::ser::Serialize,
+{
+    fn from(opts: ListConfigsOptions<T>) -> Self {
+        crate::query_parameters::ListConfigsOptionsBuilder::default()
+            .filters(
+                &opts
+                    .filters
+                    .into_iter()
+                    .map(|(k, v)| (k.into(), v.into_iter().map(T::into).collect()))
+                    .collect(),
+            )
+            .build()
+    }
+}
+
+/// Parameters used in the [Update Config API](Docker::update_config())
+///
+/// ## Examples
+///
+/// ```rust
+/// use bollard::config::UpdateConfigOptions;
+///
+/// UpdateConfigOptions{
+///     version: 1234,
+///     ..Default::default()
+/// };
+/// ```
+#[derive(Debug, Copy, Clone, Default, PartialEq, Serialize)]
+#[deprecated(
+    since = "0.19.0",
+    note = "use the OpenAPI generated bollard::query_parameters::UpdateConfigOptions and associated UpdateConfigOptionsBuilder"
+)]
+pub struct UpdateConfigOptions {
+    /// The version number of the config object being updated. This is required to avoid conflicting writes. This version number should be the value as currently set on the config before the update.
+    pub version: u64,
+}
+
+impl From<UpdateConfigOptions> for crate::query_parameters::UpdateConfigOptions {
+    fn from(opts: UpdateConfigOptions) -> Self {
+        crate::query_parameters::UpdateConfigOptionsBuilder::default()
+            .version(opts.version as i64)
+            .build()
+    }
+}
+
+impl Docker {
+    /// ---
+    ///
+    /// # List Configs
+    ///
+    /// Returns a list of configs.
+    ///
+    /// # Arguments
+    ///
+    ///  - Optional [ListConfigsOptions](ListConfigsOptions) struct.
+    ///
+    /// # Returns
+    ///
+    ///  - Vector of [Config](Config), wrapped in a Future.
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// # use bollard::Docker;
+    /// # let docker = Docker::connect_with_http_defaults().unwrap();
+    /// use bollard::config::ListConfigsOptions;
+    ///
+    /// use std::collections::HashMap;
+    /// use std::default::Default;
+    ///
+    /// let mut filters = HashMap::new();
+    /// filters.insert("label", vec!["config-label=label-value"]);
+    ///
+    /// let options = Some(ListConfigsOptions{
+    ///     filters,
+    ///     ..Default::default()
+    /// });
+    ///
+    /// docker.list_configs(options);
+    /// ```
+    pub async fn list_configs(
+        &self,
+        options: Option<impl Into<crate::query_parameters::ListConfigsOptions>>,
```

Comment:
Let's just use `crate::query_parameters::ListConfigsOptions` and avoid the deprecation warning and the alternative struct implementation. In the `v0.20` version I want to remove all these deprecations. We may in future also re-export the external crate references so people don't need to import a specific version of `bollard-stubs`, but I haven't checked how we can best do that.